lib/Ast_to_Mir.ml

```diff
@@ -160,9 +162,6 @@ let rec trans_stmt {Ast.stmt_typed; stmt_typed_loc; _} =
   in
```

Comment:
Apologies about the comment in the wrong place, but it won't let me put it where I want it: the translation of VarDecls seems not quite right to me.

We do not want the adtype to be autodiffable if we are in the transformed data or generated quantities block, right?

Also, in case we are a global variable, do we want to generate an MIR statement for that at all? Or do we just want to put it in the global var map?

Reply:
Globals should already be removed from the MIR statements with this code. Not sure what to do about autodiffable - I enjoy that almost all of the passes are not context sensitive like this yet. Maybe we need to do a pass over transformed data and gen quant to fix the vardecls, then... Not sure the best way to organize that.